### PR TITLE
Onewifi integration support bpi

### PR DESCRIPTION
**Pull request by kpandu987**

### CODE REVIEW

**Review by amarnathhullur — APPROVED** [review]

File: platform/banana-pi/platform.c [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);  
+    /*password is not sensitive,won't grant access to real devices*/ 
+    wifi_nvram_defaultRead("bpi_wifi_password",password);
+    return 0;
+}
+
+int platform_get_ssid_default(char *ssid, int vap_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);   
+    sprintf(ssid,"BPI_RDKB-AP%d",vap_index);
```
Comment (manigandanj) [inline platform/banana-pi/platform.c:151]: Need to make sure enough space available in buffer.

Comment (kpandu987) [reply platform/banana-pi/platform.c]: Taken care

File: platform/banana-pi/platform.c [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);
+    /*password is not sensitive,won't grant access to real devices*/ 
+    wifi_nvram_defaultRead("bpi_wifi_password",l_password);
+    return 0;
+}
+
+int nvram_get_current_ssid(char *l_ssid, int vap_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__); 
+    sprintf(l_ssid,"BPI_RDKB-AP%d",vap_index);
```
Comment (manigandanj) [inline platform/banana-pi/platform.c:187]: please use snprintf

File: platform/banana-pi/platform.c [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+int wifi_nvram_defaultRead(char *in,char *out);
+int _syscmd(char *cmd, char *retBuf, int retBufSize);
+
+int wifi_nvram_defaultRead(char *in,char *out)
+{
+    char buf[MAX_BUF_SIZE]={'\0'};
+    char cmd[MAX_CMD_SIZE]={'\0'};
+    char *position;
+
+    sprintf(cmd,"grep '%s=' /nvram/wifi_defaults.txt",in);
```
[the remark this answers is not in the document]
Comment (kpandu987) [reply platform/banana-pi/platform.c]: taken care

File: platform/banana-pi/platform.c (hunk 2 above)
Comment (kpandu987) [reply platform/banana-pi/platform.c]: taken care

File: src/wifi_hal_nl80211.c [hunk 4]
```diff
@@ -4837,30 +4846,42 @@ static int wiphy_dump_handler(struct nl_msg *msg, void *arg)
...
     for (j = 0; j < g_wifi_hal.num_radios; j++)
     {
         if (strcmp(g_wifi_hal.radio_info[j].name, nla_get_string(tb[NL80211_ATTR_WIPHY_NAME])) == 0) {
+            wifi_hal_dbg_print("%s:%d: Returning phy:%s already configured earlier\n",
+                __func__, __LINE__, g_wifi_hal.radio_info[j].name);
             return NL_SKIP;
         }
     }
 #endif
+    gnlh = nlmsg_data(nlmsg_hdr(msg));
```
Comment (amarnathhullur) [inline src/wifi_hal_nl80211.c:4882]: This 3 lines 4882 -4884 should go above the #if !defined(VNTXER5_PORT) as per the coverity error https://jenkins.ccp.xcal.tv/jenkins/job/TECHNICOLORXB8-Yocto-Pipeline/23313/artifact/build-images/coverity_dir/html/1/65_wifi_hal_nl80211.c.html#error

Comment (kpandu987) [reply src/wifi_hal_nl80211.c]: Done

**Pull request merged**